plone/app/redirector/browser.py

```diff
@@ -84,7 +84,17 @@ def attempt_redirect(self):
         # some analytics programs might use this info to track
         if query_string:
             url += "?" + query_string
-        self.request.response.redirect(url, status=301, lock=1)
+
+        # Answer GET requests with 302 (Found). Every other method will be answered
```

Comment:
As for switching the response type for non-GET to ones that indicate that the client should NOT switch methods, I definitely agree, since I already did that in `plone.rest` as you noticed 😉 

As for switching from permanent (301, 308) to temporary (302, 307) types of redirects: I do agree with your rationale - Plone doesn't give any guarantees about permanency of the new URL, so it could be reused for something else the next day. So temporary is semantically correct.

But, a side effect of that is, that according to [MDN - 301 Moved Permanently](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/302), this might possibly lead to detrimental effects in terms of SEO:

> A browser redirects to this page but search engines don't update their links to the resource (in 'SEO-speak', it is said that the 'link-juice' is not sent to the new URL).

So, for private intranets this may not matter, but for large public Plone sites with lots of redirects in place this could be a disadvantageous change. But honestly, I'm just not familiar enough with SEO aspects to judge whether this ("link-juice", page-rank, ...) even is relevant today. Just a point to consider.